http-clients/aws-crt-client/src/test/java/software/amazon/awssdk/http/crt/internal/CrtRequestExecutorTest.java

```diff
@@ -111,6 +122,23 @@ public void execute_AcquireConnectionFailure_shouldAlwaysWrapIOException() {
         assertThatThrownBy(executeFuture::join).hasCauseInstanceOf(IOException.class).hasRootCause(exception);
     }
 
+    @ParameterizedTest
+    @MethodSource("mappedExceptions")
+    public void execute_AcquireConnectionFailure_shouldAlwaysBeInstanceOfIOException(Entry<Integer, Class<? extends Throwable>> entry) {
```

Comment:
Can we add tests to CrtAsyncRequestExecutor as well?

Reply:
@zoewangg made the requested changes.